models/database.go

```diff
@@ -435,6 +435,20 @@ var databaseSchema = [][]string{
 			UNIQUE (name)
 		)`,
 	},
+	28: {
+		`ALTER TABLE services
+			DROP CONSTRAINT services_external_group_check`,
+
+		// Only service with type external or haproxy can have non empty value of group.
+		`ALTER TABLE services
+			ADD CONSTRAINT services_external_group_check_with_haproxy CHECK (
+				((service_type <> '` + string(ExternalServiceType) + `' OR service_type <> '` + string(HAProxyServiceType) + `') AND external_group = '')
```

Comment:
```suggestion
				((service_type <> '` + string(ExternalServiceType) + `' AND service_type <> '` + string(HAProxyServiceType) + `') AND external_group = '')
```
Shouldn't be here AND?

Reply:
@askomorokhov 
No, it is inside brackets. (external || haproxy) && empty group.
Conflicts are resolved.
Forget about API tests, not meantion in ticket, but I can write them.